### ISSUE TITLE
Map with polar stereographic has empty gap

### ISSUE THREAD

**Issue by lee1043** [UV-CDAT/uvcdat#2122]
Map with polar stereo-graphic shows empty gap for fixed column which not caused by lack of data. Below plot is coming from one dataset and gap located at different longitude. 

![screen shot 2016-09-08 at 4 04 06 pm](https://cloud.githubusercontent.com/assets/17091564/18369943/e542db70-75dd-11e6-905b-f3869010d8fc.png)

Is this bug? or is there any alternative way to fill this gap?

I have used following script:

```import cdms2 as cdms
import vcs
import cdutil

def main():
  f = cdms.open('/work/lee1043/cdat/sample_input/clt.nc')
  d = f('clt')
  plot(d, 1)

  region_selector = cdutil.region.domain(longitude=(-180.,180.))
  d2 = d(region_selector)
  plot(d2, 2)

def plot(d, n):
  x=vcs.init()
  iso = x.createisofill()
  p = vcs.createprojection()
  p.type = int('-3')
  iso.projection = p
  xtra = {}
  xtra['latitude'] = (90.0,0.0)
  d = d(**xtra)
  x.plot(d,iso)
  x.png('test'+str(n)+'.png')

main()```

**Comment by lee1043** [comment]
@charlesdoutriaux Have any idea or suggestion?

**Comment by doutriaux1** [comment]
@lee1043 you need to read extra data to close the gap longitude=(-180,182) should do.

**Comment by lee1043** [comment]
@doutriaux1 That's sounds nice trick and helps to narrow the gap. But I will also need to use my region_selector for calculation, not only for the plot. I've tried various bound options such as ccb, ccn, ooe, oon, etc., but they were not so helpful. 

![screen shot 2016-09-08 at 4 14 29 pm](https://cloud.githubusercontent.com/assets/17091564/18370114/6f459be0-75df-11e6-9729-29ae494d2c3b.png)

**Comment by doutriaux1** [comment]
@lee1043 try to set datawc_X1/x2 to 0/360 (-180,180) it might help as well.

**Comment by doutriaux1** [comment]
@danlipsa any hints for this?

**Comment by lee1043** [comment]
@doutriaux1 Thanks but I have checked 0/360 or -180/180 is not helpful for this case.

**Comment by danlipsa** [comment]
@lee1043 @doutriaux1 Try (-180, 185). That will close the gap. The original data goes from (-180, 175) so a gap is appropriate. Not sure why (-180, 180) does not close it - I thought it should have. We'll have to look closer to see why.

**Comment by lee1043** [comment]
@danlipsa @doutriaux1 Is there any option that let the program know the data is cyclic so it can close the gap when plotting?

**Comment by doutriaux1** [comment]
hum... did youtrytweaking datwc_1/x2?

**Comment by lee1043** [comment]
@doutriaux1 I have tried adding datawc_x1 and datawc_x2 but it was not so helpful, still the gap is  existing.

**Comment by lee1043** [comment]
Revisiting the issue with willingness of following up...

**Comment by doutriaux1** [comment]
@lee1043 does extending the data at plot time work?
```python
x = vcs.init()
x.plot(data(longitude=(-180,185)))
```

**Comment by lee1043** [comment]
@doutriaux1 I know that could be a temporary solution but wish that could be more generalized without having special treatment.

**Comment by lee1043** [comment]
@doutriaux1 this is interesting... I learned the gap only appears for isofill, but not for boxfill.

- boxfill:
![test_simple1](https://user-images.githubusercontent.com/17091564/41625542-fdb2e4ee-73cd-11e8-8e22-1998cb9a8cfd.png)

- isofill:
![test_simple2](https://user-images.githubusercontent.com/17091564/41625550-0837e0d6-73ce-11e8-8473-edf71ef93e97.png)

- Plotting code: [plot_test_simple.py.txt](https://github.com/CDAT/cdat/files/2117016/plot_test_simple.py.txt)
```
import cdms2, vcs
f=cdms2.open('NAM_psl_EOF1_DJF_obs_1900-2005.nc')
d=f('eof')
d.getLongitude()

x=vcs.init()
gm=vcs.createboxfill()

gm.projection='polar'
x.plot(d,gm)
x.png('test_simple1.png')

gm=vcs.createisofill()
gm.projection='polar'

x.clear()
x.plot(d,gm)
x.png('test_simple2.png')
```

- Input data: [NAM_psl_EOF1_DJF_obs_1900-2005.nc.txt](https://github.com/CDAT/cdat/files/2117019/NAM_psl_EOF1_DJF_obs_1900-2005.nc.txt)
(rename to .nc, removing .txt at the end)

**Comment by lee1043** [comment]
Another question: Can the stitches along the 0 lon (probably latitude tick label?) be removed?

**Comment by lee1043** [comment]
@doutriaux1 thanks for the `ytic1` hint. I learned both 'ytic1' and 'ytic2' should be considered as `priority=0` to completely remove the stitches. What about the isofill thing? Do you have any idea? Thanks!

**Comment by doutriaux1** [comment]
I need to look into that one, I think I need to change isofill.

**Comment by jypeter** [comment]
It would indeed be nice to fix this gap problem automatically, rather than having to generate _extra longitude_ values (even if cdms2 will nicely do this for you).

I've also had to generate extra longitudes for some non polar plots in order to avoid a white vertical band on the left or right side (don't remember which side) from time to time but I don't remember for which kind of data. Oh, you can actually see this in the Gallery:
- Problem with https://uvcdat.llnl.gov/examples/Clim_DJF_200mb_wnd_ts_gph.html
- NO _side_ problem with https://uvcdat.llnl.gov/examples/Clim_JJA_precip.html

One last note: in the examples above, you plot either the North or South pole by plotting `data(latitude=(90, -90))` or `data(latitude=(-90, 90))`, but I'm not sure this trick is explained somewhere. At least I have not found an obvious _polar_ tutorial for this on the web site

**Comment by lee1043** [comment]
Agreed to @jypeter that we need tutorial for the polar stereographic map projection. 

Just for future reference, latitude can be set as `data(latitude=(90,0))` or `data(latitude=(-90,0))` for showing only northern or southern hemisphere.

**Comment by jypeter** [comment]
Good point @lee1043 ! We just need half of the data for a polar plot :-)

**Comment by doutriaux1** [comment]
@jypeter actually the "polar" plot can display all 180 degrees of latitude. I'm not sure why it was historically called a "polar" plot, in reality it is an "Albers Equidistant" projection (proj4: aeqd)

**Comment by lee1043** [comment]
@doutriaux1 Albers Equidistant is slightly different then polar projection. Difference is at where to set edge point of flat projection of globe. Albers uses virtual point vertically above polar pole while polar projection uses actual pole at surface as one side of projected flat plane. Sorry about the poor explaining in text, I may able to give better description in person when I back ;)

**Comment by doutriaux1** [comment]
@lee1043 agreed, I just think vcs `polar` projection has been miscalled since the beginning.
